test/test_compile.py

```diff
@@ -0,0 +1,57 @@
+import argparse
+import os.path
+import pytest
+
+from otk.command import compile, parser_create
+
+
+@pytest.mark.parametrize(
+    "arguments, input_data, output_data, sys_exit_code, log_message",
+    [
+        # "GENERATE" is a placeholder to append "tmp_path" from pytest
+        (
+            argparse.Namespace(input="GENERATE", output="GENERATE", target=None),
+            """otk.version: 1
+otk.target.osbuild.qcow2: { test: 1 }
+""",
+            """{
+  "test": 1,
+  "version": "2",
+  "sources": {}
+}""",
+            0,
+            None,
+        ),
+    ],
+)
+def test_compile(
```

Comment:
Probably nice to remove the sys_exit_code and log_message as they are currently unused. and/or add a negative test that just makes sure validate fails and does not leave an (invalid/empty) output file behind.